File: webapp/backend/app/schemas/team.py
"""Team Pydantic schemas."""

from pydantic import BaseModel


class Team(BaseModel):
    team_id: int
    name: str
    abbreviation: str
    city: str | None = None
    is_active: bool = True
    founded_year: int | None = None
    arena: str | None = None
    arena_capacity: int | None = None
    franchise: dict | None = None

    class Config:
        from_attributes = True


class RosterPlayer(BaseModel):
    player_id: int
    slug: str
    name: str
    position: str | None = None
    games_played: int
    games_started: int
    ppg: float


class ScheduleGame(BaseModel):
    game_id: int
    date: str
    opponent_abbrev: str
    location: str
    result: str | None = None
    team_score: int | None = None
    opponent_score: int | None = None


class TeamSeasonStats(BaseModel):
    games_played: int
    wins: int
    losses: int
    points_per_game: float | None = None
    points_allowed_per_game: float | None = None
    offensive_rating: float | None = None
    defensive_rating: float | None = None
    net_rating: float | None = None
    pace: float | None = None
    playoff_seed: int | None = None
    made_playoffs: bool | None = None

File: src/webapp/frontend/types/team.ts
export interface Team {
  team_id: number;
  name: string;
  abbreviation: string;
  city: string;
  is_active: boolean;
  founded_year?: number;
  arena?: string;
  arena_capacity?: number;
  franchise?: {
    name: string;
    founded_year: number;
  };
}

export interface TeamScheduleGame {
  game_id: number;
  date: string;
  opponent_abbrev: string;
  location: string;
  result?: string | null;
  team_score?: number | null;
  opponent_score?: number | null;
}

export interface TeamSeasonStats {
  games_played: number;
  wins: number;
  losses: number;
  points_per_game?: number | null;
  points_allowed_per_game?: number | null;
  offensive_rating?: number | null;
  defensive_rating?: number | null;
  net_rating?: number | null;
  pace?: number | null;
  playoff_seed?: number | null;
  made_playoffs?: boolean | null;
}

export interface TeamHistorySeason {
  year: number;
  wins: number;
  losses: number;
  win_pct: number;
  made_playoffs: boolean;
  playoff_round: string | null;
}
